hooks/pre/30-el8_upgrade_postgresql.rb

```diff
@@ -0,0 +1,32 @@
+def postgresql_upgrade(new_version)
+  log_and_say(:info, "Upgrading PostgreSQL to #{new_version}")
+
+  stop_services
+
+  execute!("dnf module switch-to postgresql:#{new_version} -y", false, true)
+
+  server_packages = ['postgresql', 'postgresql-server', 'postgresql-upgrade']
+  ['postgresql-contrib', 'postgresql-docs'].each do |extra_package|
+    if execute("rpm -q #{extra_package}", false, false)
+      server_packages << extra_package
+    end
+  end
+
+  ensure_packages(server_packages, 'latest')
+
+  # puppetlabs-postgresql always sets data_directory in the config
+  # see https://github.com/puppetlabs/puppetlabs-postgresql/issues/1576
+  # however, one can't use postgresql-setup --upgrade if that value is set
+  # see https://bugzilla.redhat.com/show_bug.cgi?id=1935301
+  execute!("sed -i '/^data_directory/d' /var/lib/pgsql/data/postgresql.conf", false, true)
+
+  execute!("runuser -l postgres -c 'postgresql-setup --upgrade'", false, true)
```

Comment:
`postgresql-setup` itself drops privileges by calling `runuser`. Can't we rely on that?

Reply:
I was reading 
https://github.com/devexp-db/postgresql-setup/blob/5d24f93986bad5513e556e8080b7daeaa456199c/bin/postgresql-setup.in#L14 like "don't do that"

Reply:
Also, the other calls still need `runuser`, so we're not exactly saving mich :)

Reply:
But it would simplify this code a bit. I think this is easier to read:
```suggestion
  execute!('postgresql-setup --upgrade', false, true)
```

Reply:
Would it be better to introduce `execute_as!(user, command, …)` and call this as 
```ruby
execute_as('postgres', 'postgresql-setup --upgrade', false, true)
```

and the other call becomes
```ruby
execute_as!('postgres', 'vacuumdb --all --analyze-in-stages', false, true)
```

Reply:
I've implemented that now, lemme know what you think.